teuthology/util/scanner.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import yaml
+from collections import defaultdict
+from lxml import etree
+
+log = logging.getLogger(__name__)
+
+
+class Scanner():
+    def __init__(self, remote=None) -> None:
+        self.summary_data = []
+        self.remote = remote
+
+    def scan_all_files(self, path_regex: str):
```

Comment:
maybe include expected return types as well? since you included expected argument types?